app/src/main/java/com/itsaky/androidide/tasks/git/CloneGitTask.java

```diff
@@ -0,0 +1,195 @@
+					public void onClick(DialogInterface dia, int which) {
+						mDisposable.dispose();
+					}
+				});
+				progressDialog = materialAlertDialogBuilder.create();
+				progressDialog.show();
+
+				//createObservable
+				createObservable().subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread())
+						.subscribe(new Observer<String>() {
+							@Override
+							public void onSubscribe(Disposable disposable) {
+								Log.d(TAG, "onSubscribe: " + disposable);
+                                                                mDisposable = disposable;
+
+							}
+
+							@Override
+							public void onNext(String value) {
+								Log.d(TAG, "onNext: " + value);
+
+							}
+
+							@Override
+							public void onError(Throwable e) {
+								Log.e(TAG, "onError: ", e);
+
+								progressDialog.dismiss();
+								alertDialog.dismiss();
+
+								MaterialAlertDialogBuilder materialAlertDialogBuilder = DialogUtils.newMaterialDialogBuilder(
+										activity);
+								materialAlertDialogBuilder.setTitle(R.string.git_clone_failed);
+
+								//if file exists
+								if (e.toString().contains("org.eclipse.jgit.api.errors.JGitInternalException")) {
+									materialAlertDialogBuilder
+											.setMessage(e.toString().substring(e.toString().lastIndexOf(":") + 2));
+								}
+								//if missing url
+								else if (e.toString().contains("java.net.MalformedURLException: no protocol")) {
+									materialAlertDialogBuilder.setMessage(
+											e.toString().substring(e.toString().lastIndexOf(":") + 2) + " "
+													+ activity.getString(R.string.not_found));
+								} else {
+									//if new error found	
+									materialAlertDialogBuilder.setMessage(e.toString());
+								}
+								materialAlertDialogBuilder.setCancelable(true);
+								materialAlertDialogBuilder.setPositiveButton(R.string.okay, null);
+								materialAlertDialogBuilder.show();
+							}
+
+							@Override
+							public void onComplete() {
+								Log.d(TAG, "onComplete: ");
+								progressDialog.dismiss();
+
+								MaterialAlertDialogBuilder materialAlertDialogBuilder = DialogUtils.newMaterialDialogBuilder(
+										activity);
+								materialAlertDialogBuilder.setTitle(R.string.git_clone_successful);
+								materialAlertDialogBuilder
+										.setMessage(urlRepo + " " + activity.getString(R.string.successfully_cloned));
+								materialAlertDialogBuilder.setCancelable(true);
+								materialAlertDialogBuilder.setPositiveButton(R.string.okay, null);
+								materialAlertDialogBuilder.show();
+
+							}
+						});
+			}
+
+		});
+		materialAlertDialogBuilder.setNegativeButton(R.string.cancel, null);
+		alertDialog = materialAlertDialogBuilder.show();
+	}
+
+	public String run(String url) throws IOException {
+
+		try {
+			String urlRepo = textInputEditText.getText().toString();
+			uRL = new URL(urlRepo.toString());
+			String fileName = uRL.getFile();
+
+			//clone repo
+			Git.cloneRepository().setURI(uRL + ".git")
```

Comment:
What if the URL already contains `.git` at the end? For example, `https://github.com/AndroidIDEOfficial/AndroidIDE.git`.